src/python/pants/engine/target.py

```diff
@@ -2687,6 +2688,42 @@ class ValidatedDependencies:
     pass
 
 
+@dataclass(frozen=True)
+class DependenciesRuleActionRequest:
+    """A request to return the applicable dependency rule action for each dependency of a target."""
+
+    address: Address
+    dependencies: Addresses
+    description_of_origin: str = dataclasses.field(hash=False, compare=False)
+
+
+@dataclass(frozen=True)
+class DependenciesRuleAction:
+    """Maps all dependencies to their respective depdendecy rule action of a origin target address.
```

Comment:
s/depdendecy/dependency/ but only if you have another push for some reason. This is not worth a CI round on its own.